File: Semester 1/Class11_Computer_Contest_Lv_2_Notes_and_Homework_Nov_20-21/lvl2_hw_sol/lvl2_alice_shopping_trip.cpp
#include <bits/stdc++.h>
using namespace std;
int T, N, t, cnt; vector<int> v;
bool cmp(int x, int y){ return abs(x) < abs(y); }
int main(){
    scanf("%d %d", &T, &N);
    for(int i=0, x; i<N; i++) {
        scanf("%d", &x); v.push_back(x);
    }
    v.push_back(0);
    sort(v.begin(), v.end(), cmp);
    for(int i=1; i<v.size(); i++){
        t += abs(v[i] - v[i-1]);
        if(t > T) break;
        cnt++;
    }
    printf("%d\n", cnt);
}

File: Semester 1/Class11_Computer_Contest_Lv_2_Notes_and_Homework_Nov_20-21/lvl2_hw_sol/lvl2_alice_shopping_trip.py
t, n = map(int, input().split())
num = []
for i in range(n):
    num.append(int(input()))
num.sort(key=lambda x: abs(x))
cur, cnt = 0, 0
for i in range(n):
    if abs(num[i] - cur) <= t:
        t -= abs(num[i] - cur)
        cnt += 1
        cur = num[i]
    else:
        break
print(cnt)